### DISCUSSION TITLE
DNS Server asyncio.start_server(dns_req, "0.0.0.0", 53)

### DISCUSSION THREAD

**Discussion by SL2021-Dev** [micropython/micropython#14153] [Using MicroPython]
I use asyncio.start_server(web_req, "0.0.0.0", 80) with no issues !
But trying to capture a DNS request with... asyncio.start_server(dns_req, "0.0.0.0", 53)
never fires... async def dns_req(reader, writer):

code...
`
import asyncio, network

ap = network.WLAN(network.AP_IF)
ap.active(True)
ap.config(essid='Serve', password='12345678', authmode=network.AUTH_WPA_WPA2_PSK)

async def dns_req(reader, writer):
    request = await reader.readline()
	print(f'Received request: {request}')

async def main():
    asyncio.create_task(asyncio.start_server(dns_req, "0.0.0.0", 53))

loop = asyncio.get_event_loop()
loop.create_task(main())
loop.run_forever()
`

Can be due to UDP ???

**Comment by karfas** [comment]
What about **reading** the *fine* manual, only a google search away ?

Quoting from https://docs.micropython.org/en/latest/library/asyncio.html:

> asyncio.start_server(callback, host, port, backlog=5, ssl=None)[¶](https://docs.micropython.org/en/latest/library/asyncio.html#asyncio.start_server)
> 
>     Start a **TCP server** on the given host and port. The callback will be called with incoming, accepted connections, and be passed 2 arguments: reader and writer streams for the connection.
> 
>     If ssl is a [ssl.SSLContext](https://docs.micropython.org/en/latest/library/ssl.html#ssl.SSLContext) object, this context is used to create the transport.
> 
>     Returns a [Server](https://docs.micropython.org/en/latest/library/asyncio.html#asyncio.Server) object.
> 
>     This is a coroutine.

Besides that, what do you expect `readline()` to return ? Classic DNS consists of binary UDP packets...

**Comment by SL2021-Dev** [comment]
I know where the manual is SA...
I donn't think he asyncio.start_server() function in MicroPythoncan directly support UDP connections.

**Reply by karfas** [reply]
Fine. Then how do you interpret the "starts a **TCP** server" from these docs ?

Of course you are right, an UDP server is not supported by `start_server()`.

**Comment by sophiedegran** [comment]
Hi
Have a look at awesome-micropython.com they're several ports for dns server on micropython and they work fine
